src/components/Modals/Send/hooks/useSendDetails/useSendDetails.tsx

```diff
@@ -290,70 +296,112 @@ export const useSendDetails = (): UseSendDetailsReturnType => {
     }
   }
 
-  const inputHandler = useCallback(
+  /**
+   * handleInputChange
+   *
+   * Determines the form's state from input by onChange event.
+   * Valid inputs:
+   * - Non-empty numeric values including zero
+   * Error states:
+   * - Insufficient funds - give > have
+   * - Empty amount - input = ''
+   * - Not enough native token - gas > have
+   */
+  const handleInputChange = useCallback(
     async (inputValue: string) => {
-      setLoading(true)
       setValue(SendFormFields.SendMax, false)
+
       const key =
         fieldName !== SendFormFields.FiatAmount
           ? SendFormFields.FiatAmount
           : SendFormFields.CryptoAmount
-      if (inputValue === '') {
-        // Don't show an error message when the input is empty
-        setValue(SendFormFields.AmountFieldError, '')
-        setLoading(false)
-        // Set value of the other input to an empty string as well
-        setValue(key, '')
-        return
-      }
-      const amount =
-        fieldName === SendFormFields.FiatAmount
-          ? bnOrZero(bn(inputValue).div(price)).toString()
-          : bnOrZero(bn(inputValue).times(price)).toString()
 
-      setValue(key, amount)
+      try {
+        if (inputValue === '') {
+          // Cancel any pending requests
+          debouncedEstimateFormFees.cancel()
+          // Don't show an error message when the input is empty
+          setValue(SendFormFields.AmountFieldError, '')
+          // Set value of the other input to an empty string as well
+          setValue(key, '') // TODO: this shouldnt be a thing, using a single amount field
+          return
+        }
 
-      let estimatedFees
+        const amount =
+          fieldName === SendFormFields.FiatAmount
+            ? bnOrZero(bn(inputValue).div(price)).toString()
+            : bnOrZero(bn(inputValue).times(price)).toString()
 
-      try {
-        estimatedFees = await estimateFormFees()
-        setValue(SendFormFields.EstimatedFees, estimatedFees)
-      } catch (e) {
-        setValue(SendFormFields.AmountFieldError, 'common.insufficientFunds')
-        setLoading(false)
+        setValue(key, amount)
 
-        throw e
-      }
+        // TODO: turn to actual validators
+        let estimatedFees
+        let hasEnoughNativeTokenForGas = false
+        let hasValidBalance = false
+
+        const { cryptoAmount } = getValues()
+        hasValidBalance = cryptoHumanBalance.gte(cryptoAmount)
+
+        if (!hasValidBalance) {
+          throw new Error('common.insufficientFunds')
+        }
+
+        try {
```

Comment:
we can do an IIFE here for estimatedFees

Reply:
Done.